### PR TITLE
[CMake] Updates to allow inclusion using FetchContent

### PR DESCRIPTION
**Pull request by bnbarham**
Various updates that allow swift-syntax to be included using FetchContent.

Mostly this is just not replacing `target_link_libraries` since that is a global replacement, but we also now use a couple of variables from the swift if they're set (eg. `SWIFT_HOST_LIBRARIES_DEST_DIR` and `SWIFT_HOST_MODULE_TRIPLE`).

### CODE REVIEW

File: cmake/modules/AddSwiftHostLibrary.cmake [hunk 1]
```diff
@@ -6,22 +6,44 @@
...
+      )
+    target_sources(${TARGET} PRIVATE
+      ${CMAKE_CURRENT_BINARY_DIR}/forced-${DEPENDENCY}-dep.swift
+    )
+  endforeach()
+endfunction()
+
 # Add a new host library with the given name.
-function(add_swift_host_library name)
+function(add_swift_syntax_library name)
```
Comment (etcwilde) [inline cmake/modules/AddSwiftHostLibrary.cmake:35]: Is this function only being used for SwiftSyntax, or will we want to pull in other things this way too?

Comment (etcwilde) [reply cmake/modules/AddSwiftHostLibrary.cmake:35]: Oh, this is the swift-syntax repo, don't mind me!

Comment (bnbarham) [reply cmake/modules/AddSwiftHostLibrary.cmake:35]: It's only for swift-syntax, though theoretically we could replace the swift functions with functions defined here instead. 🤷

**Comment by bnbarham** [comment]
Updated to remove export if not a top level project + add to SWIFT_EXPORTS otherwise.

**Review by DougGregor — APPROVED** [review]
This all looks reasonable to me

File: Sources/SwiftIDEUtils/CMakeLists.txt [hunk 2]
```diff
@@ -6,12 +6,13 @@
 # See http://swift.org/LICENSE.txt for license information
 # See http://swift.org/CONTRIBUTORS.txt for Swift project authors
 
-add_swift_host_library(SwiftIDEUtils
+add_swift_syntax_library(SwiftIDEUtils
+  generated/SyntaxClassification.swift
```
[the remark this answers is not in the document]
Comment (bnbarham) [reply Sources/SwiftIDEUtils/CMakeLists.txt:10]: No, this will have been a bad conflict resolution

**Pull request closed without merging**